packages/elasticsearch-plugin/src/indexer.controller.ts

```diff
@@ -549,12 +466,12 @@ export class ElasticsearchIndexerController implements OnModuleInit, OnModuleDes
                     updatedProductVariants.forEach(v => (v.enabled = false));
                 }
                 Logger.verbose(`Updating Product (${productId})`, loggerCtx);
-                if (updatedProductVariants.length) {
-                    operations.push(...(await this.updateVariantsOperations(updatedProductVariants)));
+                const languageVariants: LanguageCode[] = [];
+                languageVariants.push(...product.translations.map(t => t.languageCode));
+                for (const variant of product.variants) {
+                    languageVariants.push(...variant.translations.map(t => t.languageCode));
```

Comment:
I have been testing the latest commit with my real project, and keep running into a "Maximum call stack size exceeded" error which causes the `reindex` job to fail.

I have slowly traced it down to this line, I _think_ which might be causing the issue. The job keeps failing on the same product, which has 139 variants. After this line has executed, the `languageVariants` array consists of `'en'` x 140! (1 for the product + 139 for each variant).

This is then causing 140x the amount of work in line 493 to execute **19,460** times, rather than 139 times. I think this is related to the call stack error.

I believe the missing part is adding the `unique()` helper function to only use unique values in the `languageVariants` array.

Reply:
Yeah, in previous version of this PR there was not any problem. I also caught this error today
```
worker_1  | verbose  [ElasticsearchPlugin] Updating 1 Products
worker_1  | verbose  [ElasticsearchPlugin] Deleting 1 Product (id: 5990)
worker_1  | verbose  [ElasticsearchPlugin] Deleting 312 ProductVariants
worker_1  | verbose  [ElasticsearchPlugin] Updating Product (5990)
worker_1  | error  [Vendure Worker] {}
```

Reply:
Fixed. Successfully rebuild index on my dev